cli/src/cli.ts

```diff
@@ -201,14 +238,36 @@ function parseArgs(): Args {
   // Add back any arguments that came after --
   const finalArgs = [...remainingArgs, ...postArgs];
 
-  // Validate that config and server are provided together
-  if (
-    (options.config && !options.server) ||
-    (!options.config && options.server)
-  ) {
-    throw new Error(
-      "Both --config and --server must be provided together. If you specify one, you must specify the other.",
+  // Validate config and server options
+  if (!options.config && options.server) {
+    throw new Error("--server requires --config to be specified");
+  }
+
+  // If config is provided without server, try to auto-select
+  if (options.config && !options.server) {
+    const configContent = fs.readFileSync(
+      path.isAbsolute(options.config)
+        ? options.config
+        : path.resolve(process.cwd(), options.config),
+      "utf8",
     );
+    const parsedConfig = JSON.parse(configContent);
+    const servers = Object.keys(parsedConfig.mcpServers || {});
+
+    if (servers.includes("default-server")) {
```

Comment:
Maybe just pass this name as a flag, and w/o it only auto select if there's 1 server?

Reply:
Amended.